src/query/joining.rs

```diff
@@ -50,14 +51,17 @@ impl NestedQuery {
 /// Has Child query
 #[derive(Debug, Default, Serialize)]
 pub struct HasChildQuery {
+    #[serde(rename="type")]
     doc_type: String,
     query: Query,
     #[serde(skip_serializing_if="ShouldSkip::should_skip")]
     score_mode: Option<ScoreMode>,
     #[serde(skip_serializing_if="ShouldSkip::should_skip")]
     min_children: Option<u64>,
     #[serde(skip_serializing_if="ShouldSkip::should_skip")]
-    max_children: Option<u64>
+    max_children: Option<u64>,
+    #[serde(skip_serializing_if="ShouldSkip::should_skip")]
+    inner_hits: Option<Value>
```

Comment:
Preferably this would be a concrete type rather than `Value`, but the ElasticSearch documentation is... less than comprehensive on this matter.  This seems to be the best reference: https://www.elastic.co/guide/en/elasticsearch/reference/1.5/search-request-inner-hits.html unless there's a better one somewhere?

Reply:
The link I sent you in the other comment is similar to this one - Although I agree that a concrete type would be way preferable, I find not using a `Value` quite a hard way here 😞

Reply:
Because of Inner Hits potentially returning more than one query?  Yes, that's true.

We could have a hybrid option as there is some predictable structure.  E.g. the `"total"` and `"_id"` fields, it's only really the `"_source"` field that needs to be a `Value`?